spring-cloud-starter-single-step-batch-job/src/test/java/org/springframework/cloud/task/batch/autoconfigure/kafka/KafkaItemReaderAutoConfigurationTests.java

```diff
@@ -0,0 +1,240 @@
+				.withConfiguration(
+						AutoConfigurations.of(PropertyPlaceholderAutoConfiguration.class,
+								BatchAutoConfiguration.class,
+								SingleStepJobAutoConfiguration.class,
+								KafkaItemReaderAutoConfiguration.class))
+				.withPropertyValues("spring.batch.job.jobName=job",
+						"spring.batch.job.stepName=step1", "spring.batch.job.chunkSize=5",
+						"spring.kafka.consumer.bootstrap-servers="
+								+ embeddedKafkaBroker.getBrokersAsString(),
+						"spring.kafka.consumer.group-id=1",
+						"spring.batch.job.kafkaitemreader.name=kafkaItemReader",
+						"spring.batch.job.kafkaitemreader.topic=" + topicName,
+						"spring.kafka.consumer.value-deserializer="
+								+ JsonDeserializer.class.getName());
+		Date startTime = new Date();
+		applicationContextRunner.run((context) -> {
+			JobLauncher jobLauncher = context.getBean(JobLauncher.class);
+
+			Job job = context.getBean(Job.class);
+
+			ListItemWriter itemWriter = context.getBean(ListItemWriter.class);
+
+			JobExecution jobExecution = jobLauncher.run(job, new JobParameters());
+
+			JobExplorer jobExplorer = context.getBean(JobExplorer.class);
+
+			while (jobExplorer.getJobExecution(jobExecution.getJobId()).isRunning()) {
+				Thread.sleep(1000);
+			}
+			Date endTime = new Date();
+			long seconds = (endTime.getTime() - startTime.getTime()) / 1000;
+			assertThat(seconds).isGreaterThanOrEqualTo(30);
+			basicValidation(itemWriter);
+		});
+	}
+
+	private void basicValidation(ListItemWriter itemWriter) {
+		List<Map<Object, Object>> writtenItems = itemWriter.getWrittenItems();
+		assertThat(writtenItems.size()).isEqualTo(4);
+		List<Object> results = new ArrayList<>();
+		for (int i = 0; i < 4; i++) {
+			results.add(writtenItems.get(i).get("first_name"));
+		}
+
+		assertThat(results).contains("jane", "john", "susan", "jim");
+	}
+
+	private void populateSingleTopic(String topic) {
+		Map<String, Object> configps = new HashMap<>(
+				KafkaTestUtils.producerProps(embeddedKafkaBroker));
+		Producer<String, Object> producer = new DefaultKafkaProducerFactory<>(configps,
+				new StringSerializer(), new JsonSerializer<>()).createProducer();
+		Map<Object, Object> testMap = new HashMap<>();
+		testMap.put("first_name", "jane");
+		producer.send(new ProducerRecord<>(topic, "my-aggregate-id", testMap));
+		testMap = new HashMap<>();
+		testMap.put("first_name", "john");
+		producer.send(new ProducerRecord<>(topic, "my-aggregate-id", testMap));
+		testMap = new HashMap<>();
+		testMap.put("first_name", "susan");
+		producer.send(new ProducerRecord<>(topic, "my-aggregate-id", testMap));
+		testMap = new HashMap<>();
+		testMap.put("first_name", "jim");
+		producer.send(new ProducerRecord<>(topic, "my-aggregate-id", testMap));
+		producer.flush();
+		producer.close();
+	}
+
+	@EnableBatchProcessing
+	@Configuration
+	public static class CustomMappingConfiguration {
+		@Bean
+		public ListItemWriter<Map> itemWriter() {
```

Comment:
This could be:

```suggestion
		public ListItemWriter<Map<Object, Object>> itemWriter() {
```

to avoid the `Raw use of parameterized class 'Map'` warning.